api/src/controllers/images.ts

```diff
@@ -366,8 +367,13 @@ export async function isCached(
 	if (headers["if-modified-since"]) {
 		const ifModifiedSince = headers["if-modified-since"];
 		let lastModified: Date | undefined;
+		const stat = await getFile(filePath).stat();
 		try {
-			lastModified = (await stat(filePath)).mtime;
+			if (stat as S3Stats) {
```

Comment:
I'm pretty sure this won't work. typescript types are just syntactic sugar and as won't return undefined if it's the wrong type.

you could do something like

```typescript
if ("lastModified" in stat)
```

instead

Reply:
Thanks! I was not aware of this. I read up a little and I _think_ I'm doing it the idiomatic way now. PTAL and I'll update again if I'm wrong.